### DISCUSSION TITLE
Access RemoteAddr inside of TraceLayer

### DISCUSSION THREAD

**Discussion by mystborn** [tower-rs/tower-http#428] [Q&A] [answered]
I'm currently building a web server with Axum, and I''m using `TraceLayer::new_for_http` to log incoming requests. Is it possible to access the remote IP address from inside of `make_span_with`?

**Comment by davidpdrsn** [comment]
You can get https://docs.rs/axum/0.6.20/axum/extract/struct.ConnectInfo.html from the request extensions.

**Reply by mystborn** [reply]
I don't think it's possible to get an extractor from the make_span_with function. I do make sure to serve with `into_make_service_with_connect_info`, but there's no way to get the `ConnectInfo` from `make_span_with`. `Request.into_parts` can't be used either because it would consume the borrowed resource.

**Reply by davidpdrsn** [reply, marked as answer]
You don’t need the Extension extractor. You can just do `request.extensions().get::<ConnectInfo<SocketAddr>>()`
